src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -2951,6 +3245,24 @@ private QueryOperationResponse ContinuePage(QueryOperationResponse response, obj
             return response;
         }
 
+        private Task<QueryOperationResponse> ContinuePageAsync(QueryOperationResponse response, object entity, string propertyName, CancellationToken cancellationToken)
+        {
+            var continuation = response.GetContinuation();
+            if (continuation != null)
+            {
+                IAsyncResult beginLoadPropertyResult = this.BeginLoadProperty(entity, propertyName, continuation, null, null);
+                cancellationToken.Register(() => this.CancelRequest(beginLoadPropertyResult));

```

Comment:
@KenitoInc Curious what would happen here for `cancellationToken` equals `CancellationToken.None`